### PR TITLE
feat: Create Permits definitions for snip20 and snip721 #114

### PR DESCRIPTION
**Pull request by David-Sod**

### CODE REVIEW

File: src/contracts/definitions/snip-721-def.ts [hunk 1]
```diff
@@ -513,6 +516,179 @@ export const snip721Def: ContractDefinition = {
...
+
+    getVerifyTransferApproval({ permit }: Context, token_ids: Array<string>) {
+      const query = { verify_transfer_approval: { token_ids } };
+
+      return { with_permit: { query, permit } };
+    },
+  },
+};
+
+export const snip721DefPermit = extendContract(snip721Def, snip721Permit);
```
Comment (luis-stakeordie) [inline src/contracts/definitions/snip-721-def.ts:690]: I would say that as a good practice to add the `Def` suffix to the name of a contract definition. So in this case should be `snip721PermitDef`.

File: src/contracts/definitions/snip-721-def.ts [hunk 2]
```diff
@@ -513,6 +516,179 @@ export const snip721Def: ContractDefinition = {
   },
 };
 
+const snip721Permit: ContractDefinition = {
```
Comment (luis-stakeordie) [inline src/contracts/definitions/snip-721-def.ts:519]: Check the other comment, but this should be `snip721BasePermitDef`

**Review by luis-stakeordie — CHANGES_REQUESTED** [review]
Check this comments out @David-Sod

**Comment by David-Sod** [comment]
I renamed the permits definitions as you mentioned

**Review by luis-stakeordie — APPROVED** [review]
Thanks

**Pull request merged**